app/Http/Controllers/Dormitory/Printing/PrintAccountController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace App\Http\Controllers\Dormitory\Printing;
+
+use App\Http\Controllers\Controller;
+use App\Mail\ChangedPrintBalance;
+use App\Models\Checkout;
+use App\Models\PaymentType;
+use App\Models\Semester;
+use App\Models\Transaction;
+use App\Models\User;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Mail;
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Container\ContainerExceptionInterface;
+
+class PrintAccountController extends Controller
+{
+    /**
+     * Updates balance of a `PrintAccount`.
+     * This can be done in two ways: either by topping up the account by giving money to an admin
+     * or by transfering money from one account to the other.
+     */
+    public function update(Request $request)
+    {
+        $request->validate([
+            'amount' => 'required|integer',
+            'user' => 'required|exists:users,id', // Normally this would be a path parameter for the PrintAccount, but we can't do that because of the limitations of blade templates
+            'other_user' => 'nullable|exists:users,id',
+        ]);
+
+        $printAccount = User::find($request->get('user'))->printAccount;
+
+        // If user can not even transfer balance, we can stop here
+        if (user()->cannot('transferBalance', $printAccount)) {
+            abort(403);
+        }
+
+        $otherAccount = $request->other_user ? User::find($request->get('other_user'))->printAccount : null;
+
+        // This is a transfer between accounts
+        if ($otherAccount !== null) {
+            // Cannot transfer to yourself
+            if ($otherAccount->user_id === $printAccount->user_id) {
```

Comment:
Is $otherAccount and $printAccount the same PHP object? I'm not sure. If they are different, decreasing $printAccount's balance and saving it, and then increasing $otherAccount's balance and saving it might add money to the account in the end.
Like:
```
a = 1500
b = a #1500
c = a #1500
amount = 300 
b -= amount #1200
a = b #1200
c += amount #1800
a = c #1800
```